assets/app/scripts/controllers/services.js

```diff
@@ -24,7 +25,24 @@ angular.module('openshiftConsole')
       $scope.emptyMessage = "No services to show";
       updateFilterWarning();
       console.log("services (subscribe)", $scope.unfilteredServices);
-    }));    
+    }));
+
+    watches.push(DataService.watch("routes", $scope, function(routes){
+        $scope.routes = sortRoutes(routes._data);
+        console.log("routes (subscribe)", $scope.routes);
+    }));
+
+    var sortRoutes = function(routes) {
+        var routeMap = {};
+        angular.forEach(routes, function(route){
+            if (!routeMap[route.serviceName]){
```

Comment:
we like to keep these reduced to one-liners:
<code>routeMap[route.serviceName] = routeMap[route.serviceName] || [];</code>